blueflood-core/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/RollupHandler.java

```diff
@@ -56,114 +63,145 @@
+                                final Timer.Context rollupsCalcCtx = rollupsCalcOnReadTimer.time();
+
+                                if (metricData.getValue().getData().isEmpty()) { // data completely missing for range. complete repair.
+                                    rollupsRepairEntireRange.mark();
+                                    List<Points.Point> repairedPoints = repairRollupsOnRead(metricData.getKey(), g, from, to);
+                                    for (Points.Point repairedPoint : repairedPoints) {
+                                        metricData.getValue().getData().add(repairedPoint);
+                                    }
+
+                                    if (repairedPoints.isEmpty()) {
+                                        rollupsRepairEntireRangeEmpty.mark();
+                                    }
+                                } else {
+                                    long actualStart = minTime(metricData.getValue().getData());
+                                    long actualEnd = maxTime(metricData.getValue().getData());
+
+                                    // If the returned start is greater than 'from', we are missing a portion of data.
+                                    if (actualStart > from) {
+                                        rollupsRepairedLeft.mark();
+                                        List<Points.Point> repairedLeft = repairRollupsOnRead(metricData.getKey(), g, from, actualStart);
+                                        for (Points.Point repairedPoint : repairedLeft) {
+                                            metricData.getValue().getData().add(repairedPoint);
+                                        }
+
+                                        if (repairedLeft.isEmpty()) {
+                                            rollupsRepairedLeftEmpty.mark();
+                                        }
+                                    }
+
+                                    // If the returned end timestamp is less than 'to', we are missing a portion of data.
+                                    if (actualEnd + g.milliseconds() <= to) {
+                                        rollupsRepairedRight.mark();
+                                        List<Points.Point> repairedRight = repairRollupsOnRead(metricData.getKey(), g, actualEnd + g.milliseconds(), to);
+                                        for (Points.Point repairedPoint : repairedRight) {
+                                            metricData.getValue().getData().add(repairedPoint);
+                                        }
+
+                                        if (repairedRight.isEmpty()) {
+                                            rollupsRepairedRightEmpty.mark();
+                                        }
+                                    }
+                                }
+                                retValue = true;
```

Comment:
@tilogaat Its not required per se, but because we are using a `callable` we need to return some `Object`, so I thought instead of returning Void or null, return a boolean variable indicating `success` or `failure`.